backend/src/main/kotlin/code/nebula/cipherquest/repository/entities/UserLevel.kt

```diff
@@ -11,5 +15,32 @@ data class UserLevel(
     var level: Int,
     var username: String,
     var coins: Int = 25,
+    @CreationTimestamp
+    var createdAt: OffsetDateTime = OffsetDateTime.now(),
+    @UpdateTimestamp
+    var updatedAt: OffsetDateTime = OffsetDateTime.now(),
     var terminatedAt: OffsetDateTime? = null,
-)
+    var score: Long = 0,
+) {
+    // coins -> increase by each coin twice
+    // time -> decrease for each minute after a threshold of 25 mins
+    // level -> increase - 1: 0, 2: 30, 3: 70
+    // win -> increase by 100
+    fun updateScore() {
```

Comment:
I'd move this function into `UserLevelService`, to keep the business logic outside of the DB `Entity` classes